sentry/src/main/java/io/sentry/config/PropertiesProviderFactory.java

```diff
@@ -0,0 +1,61 @@
+package io.sentry.config;
+
+import io.sentry.ILogger;
+import io.sentry.SystemOutLogger;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+@ApiStatus.Internal
+public final class PropertiesProviderFactory {
```

Comment:
This class remains untested. Since there is no easy way to mock environment variables it is tricky to test it.